### PR TITLE
Add support for block and scripts

### PR DESCRIPTION
**Pull request by wismill**
Add support for block and scripts.

### CODE REVIEW

**Comment by wismill** [comment]
Rebased & squashed

**Comment by wismill** [comment]
@harendra-kumar @adithyaov could you have a look at it?

File: unicode-data/exe/Parser/Text.hs [hunk 1]
```diff
@@ -256,6 +267,296 @@ isHangul c = n >= hangulFirst && n <= hangulLast
...
+        else error ("Unsupported character: " <> show c)
+
+genBlocksModule
+    :: Monad m
+    => String
+    -> Fold m BlockLine String
+genBlocksModule moduleName = done <$> Fold.foldl' step initial
+    where
+
+    done (blocks, defs, ranges) = let ranges' = reverse ranges in unlines
```
Comment (adithyaov) [inline unicode-data/exe/Parser/Text.hs:285]: Can probably use `Streamly.Internal.Unicode.String.str`.

**Review by adithyaov — APPROVED** [review]
Apologies for the extremely delayed review.
Overall, LGTM!

File: unicode-data/exe/Parser/Text.hs (hunk 1 above)
Comment (wismill) [reply unicode-data/exe/Parser/Text.hs:285]: I find reference to neither the module nor the function on Hackage.

**Review by harendra-kumar — CHANGES_REQUESTED** [review]
* The script file is quite big, this will also make the compiled library size big, do we want to put it into a separate package like name/name-aliases? 
* To avoid making too many packages we can possibly have just two packages, we can call the lightweight package as unicode-data-core and bundle everything including unicode-data-core in the all-inclusive unicode-data package.
* What is difference in "script" and "inScript" APIs? can `inScript` be derived from `script`? Similar question for `block` and `inBlock`.

**Comment by harendra-kumar** [comment]
@wismill I think the review got delayed because this was a wip earlier and I missed when it became ready for review. If you are not getting attention for some PR please ping me or send an email requesting review on my personal email id harendra.kumar at gmail.

File: unicode-data/lib/Unicode/Internal/Char/Blocks.hs [hunk 2]
```diff
@@ -0,0 +1,730 @@
...
+(Block(..), BlockDefinition(..), block, blockDefinition, allBlockRanges)
+where
+
+import Data.Ix (Ix)
+import GHC.Exts
+
+-- [TODO] @since
+-- | Unicode block.
+data Block
+    = Basic_Latin -- ^ @U+0000..U+007F@: Basic Latin.
```
[the remark this answers is not in the document]
Comment (wismill) [reply unicode-data/lib/Unicode/Internal/Char/Blocks.hs]: Done

**Comment by wismill** [comment]
> * The script file is quite big, this will also make the compiled library size big, do we want to put it into a separate package like name/name-aliases?
> 
> * To avoid making too many packages we can possibly have just two packages, we can call the lightweight package as unicode-data-core and bundle everything including unicode-data-core in the all-inclusive unicode-data package.

I do not have a strong opinion on this. I like the idea of `unicode-data-core` with the bare minimum and `unicode-data` with “all batteries included”. But then we should define criteria where to include APIs. Currently there are [4 packages depending on `unicode-data`](https://packdeps.haskellers.com/reverse/unicode-data).

Let’s merge this PR and discuss the split in another issue.

>  * What is difference in "script" and "inScript" APIs? can `inScript` be derived from `script`? Similar question for `block` and `inBlock`.

Fixed: `inScript s = (== s) . script`

**Comment by wismill** [comment]
Rebased

**Comment by wismill** [comment]
@harendra-kumar I splited scripts into a new package `unicode-data-scripts`.

**Pull request merged**